pkg/operator/v1helpers/helpers.go

```diff
@@ -347,3 +349,65 @@ func InjectObservedProxyIntoContainers(podSpec *corev1.PodSpec, containerNames [
 
 	return nil
 }
+
+func IsDeploymentProgressing(conditionName string, deployment *appsv1.Deployment, status *operatorv1.OperatorStatus) (bool, string) {
+	if deployment.Generation != deployment.Status.ObservedGeneration {
+		return true, "Waiting for Deployment to act on changes"
+	}
+
+	var expectedReplicas int32
+	if deployment.Spec.Replicas != nil {
+		expectedReplicas = *deployment.Spec.Replicas
+	}
+
+	if deployment.Status.AvailableReplicas < expectedReplicas {
```

Comment:
A deployment about to roll out changes may be completely available, but still not completely updated.  If we are looking at pod counts instead of the Deployment's `conditions`, I'd expect `updatedReplicas` to come into this.

Reply:
Thanks! Apparently, `AvailableReplicas` is the total number of replicas in all Deployment's ReplicaSets, *regardless* the current `Deployment.Spec`. So you're right, this check might evaluate to `false` even though new config is rolling out.

On the other hand, `UpdatedReplicas` has the number of replicas of the *newer* ReplicaSet only, the one that represents the current `Deployment.Spec`. That being said, I think we only need to check `UpdatedReplicas` here.